mne/forward/_make_forward.py

```diff
@@ -218,86 +226,145 @@ def _setup_bem(bem, bem_extra, neeg, mri_head_t, verbose=None):
-        neeg = len(picks)
-        if neeg > 0:
-            eegchs = pick_info(info, picks)['chs']
-            eegnames = [info['ch_names'][p] for p in picks]
-            logger.info('Read %3d EEG channels from %s'
-                        % (len(picks), info_extra))
+    # Find MEG channels
+    picks = pick_types(info, meg=True, eeg=False, ref_meg=False,
+                       exclude=exclude)
+
+    # Get channel info and names for MEG channels
+    nmeg = len(picks)
+    if nmeg > 0:
+        megchs = pick_info(info, picks)['chs']
+        megnames = [info['ch_names'][p] for p in picks]
+        logger.info('Read %3d MEG channels from %s'
+                    % (len(picks), info_extra))
+
+    # Get MEG compensation channels
+    if not ignore_ref:
+        picks = pick_types(info, meg=False, ref_meg=True, exclude=exclude)
+        ncomp = len(picks)
+        if (ncomp > 0):
+            compchs = pick_info(info, picks)['chs']
+            logger.info('Read %3d MEG compensation channels from %s'
+                        % (ncomp, info_extra))
+            # We need to check to make sure these are NOT KIT refs
+            if _has_kit_refs(info, picks):
+                err = ('Cannot create forward solution with KIT reference '
+                       'channels. Consider using "ignore_ref=True" in '
+                       'calculation')
+                raise NotImplementedError(err)
     else:
-        neeg = 0
-        logger.info('EEG not requested. EEG channels omitted.')
+        ncomp = 0
+
+    _print_coord_trans(info['dev_head_t'])
 
-    if neeg <= 0 and nmeg <= 0:
-        raise RuntimeError('Could not find any MEG or EEG channels')
+    # Make info structure to allow making compensator later
+    ncomp_data = len(info['comps'])
+    ref_meg = True if not ignore_ref else False
+    picks = pick_types(info, meg=True, ref_meg=ref_meg, exclude=exclude)
+    meg_info = pick_info(info, picks) if nmeg > 0 else None
 
     # Create coil descriptions with transformation to head or MRI frame
-    templates = _read_coil_defs()
-    if nmeg > 0 and ncomp > 0:  # Compensation channel information
-        logger.info('%d compensation data sets in %s'
-                    % (ncomp_data, info_extra))
+    templates = _read_coil_defs(elekta_defs=elekta_defs)
     if nmeg > 0:
-        megcoils = _create_coils(megchs, accuracy,
-                                 info['dev_head_t'], 'meg', templates)
+        megcoils = _create_coils(megchs, accuracy, info['dev_head_t'], 'meg',
+                                 templates)
         if ncomp > 0:
+            logger.info('%d compensation data sets in %s' % (ncomp_data,
+                                                             info_extra))
             compcoils = _create_coils(compchs, FIFF.FWD_COIL_ACCURACY_NORMAL,
                                       info['dev_head_t'], 'meg', templates)
+        logger.info('Head coordinate MEG coil definitions created.')
```

Comment:
Instead of indenting all this, make the conditional above `if nmeg == 0: raise RuntimeError(...)` and then all this can follow it, unindented. Same with EEG.